crates/oxc_semantic/src/node.rs

```diff
@@ -94,6 +94,15 @@ impl<'a> AstNodes<'a> {
         self.nodes.iter()
     }
 
+    /// Walk up the AST, iterating over each parent node.
+    ///
+    /// The first node produced by this iterator is the first parent of the node
+    /// pointed to by `node_id`. The last node will usually be a `Program`.
+    pub fn iter_parents(&self, node_id: AstNodeId) -> impl Iterator<Item = &AstNode<'a>> + '_ {
```

Comment:
Does the method `ancestors` work for you?

https://github.com/web-infra-dev/oxc/blob/3b9cc474e94924cfb4318edb3d82e489906ff677/crates/oxc_semantic/src/node.rs#L121-L124

Reply:
It could, but i'd need to throw away the first value and then unpack the `Option` in each iteration. Which do you think is cleaner?

Reply:
We can combine the two in a later PR, your solution seem better.

Reply:
I've gone back and forth, trying both. These two iterators should be functionally equivalent, but multiple tests fail when I use `ancestors`. I think some nodes are being skipped that shouldn't be? I'm not sure, but I'm going to leave in `iter_parents` for now.